src/sage/matrix/matrix2.pyx

```diff
@@ -9898,6 +9898,44 @@ cdef class Matrix(Matrix1):
         self.cache(key, normal)
         return normal
 
+    def is_nilpotent(self) -> bool:
+        r"""
+        Return if ``self`` is a nilpotent matrix.
+
+        A matrix `A` is *nilpotent* if there exists a positive integer
+        `k` such that `A^k = 0`. We test this by using the Cayley-Hamilton
+        theorem to see if the characteristic polynomial is `x^n = 0`.
```

Comment:
shouldn't that be `x^k = 0`?

Reply:
actually, it might be more precise to say:
```
We test this by checking whether the characteristic polynomial is a monomial, `x^n`.
The Cayley-Hamilton theorem asserts that then the minimal `k` such that `A^k = 0` equals `n`.
```

Reply:
Wot ? the minimal k can be anything between 1 and n...

Reply:
upsi, sorry, in delirium I confused minimal and characteristic polynomial.  now going back to sleep.